### DISCUSSION TITLE
从外表导入数据到目标的表，有数据丢失

### DISCUSSION THREAD

**Discussion by yclxiao** [apache/doris#14158] [A - General / Q&A]
<img width="1740" alt="image" src="https://user-images.githubusercontent.com/8273732/201020211-645cdaf2-b073-4546-a289-966bc153b01d.png">

从外表导入数据到正常的表，有数据丢失。
我创建的外表数据是1880条，但是执行完insert之后，目标表的数据是1823，少了好几十条。
我的目标表设置了id唯一。
求大佬帮助解答下，谢谢！

**Comment by zy-kkk** [comment]
在外部表查询一下 select count(distinct id) from tbl; 看下多少条呢

**Reply by yclxiao** [reply]
外表执行这个SQL是1880条

**Comment by yuanyuan8983** [comment]
Please open strict mode to see whether there is an error when inserting data：SET enable_insert_strict = true;

**Reply by yclxiao** [reply]
<img width="1416" alt="image" src="https://user-images.githubusercontent.com/8273732/201038925-e211c36c-a71c-4f76-a6ca-4225df96c82f.png">
确实是出现了导入失败的情况，字符串出现了部分乱码，而且字段长度不够，很奇怪。为什么呢

**Reply by zy-kkk** [reply]
未开启严格模式的情况下，一批数据insert会只插入符合的数据，不合理的数据直接过滤掉，你的返回结果应该会显示有warning

**Comment by yuanyuan8983** [comment]
首先查看你表的对应column的数据类型的大小，这个就是插入的数据和对应列的数据类型大小不一致导致的，可以通过alter进行调整，之后就可以正常插入数据了
